### PR TITLE
Refs #21007 - new failed call for kickstarts

### PR DESCRIPTION
**Pull request by lzap**
Core PR needed: https://github.com/theforeman/foreman/pull/5292

### CODE REVIEW

**Comment by lzap** [comment]
This needs review with testing, core PR needed.

File: provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -204,12 +209,19 @@ sed -e 's/DEFAULTKERNEL=kernel-uek/DEFAULTKERNEL=kernel/g' -i /etc/sysconfig/ker
...
-) 2>&1 | tee /root/install.post.log
-exit 0
-
+<%#
+The last post section halts Anaconda to prevent endless loop on EL7+
+%>
+%post <%= erroronfail -%>
+if test -f /tmp/foreman_built; then
+  echo "calling home: build is done!"
+  wget -q -O - --body-file=/mnt/sysimage/root/install.post.log --method GET --header="Content-Type: text/xml" --no-check-certificate <%= foreman_url('built') %>
```
Comment (iNecas) [inline provisioning_templates/provision/kickstart_default.erb]: Shouldn't the `Content-type` be `text/plain`?

Comment (lzap) [reply provisioning_templates/provision/kickstart_default.erb]: Yeah typo.

**Comment by iNecas** [comment]
The foreman PR is merged, but tests are failing here

**Comment by lzap** [comment]
Fixed tests.

**Comment by iNecas** [comment]
You think :) Seems like fedora 25-27 is not supported in pykickstart.

**Comment by ekohl** [comment]
let's see if https://github.com/theforeman/community-templates/pull/485 can fix that

**Comment by lzap** [comment]
Yeah removing from this PR and will rebase on top of that once we merge #485 with priority. Looks good there.

**Comment by timogoebel** [comment]
@lzap: What's the status here?

**Comment by lzap** [comment]
Lack of reviewers? :-) Rebased.

**Comment by mmoll** [comment]
@lzap Did the rebase go wrong? Please check the test failure :broken_heart:

**Comment by lzap** [comment]
Yep, I screwed up. Testing++ GitInteractiveHunkStaging++

**Comment by mmoll** [comment]
@lzap please rebase

**Comment by lzap** [comment]
This is how it looks like now. I am likely going to fix "indent" method to ignore last line of snippet because this breaks output all the time:

```

%post --erroronfail
if test -f /tmp/foreman_built; then
  echo "calling home: build is done!"
    if [ -x /usr/bin/curl ]; then
    /usr/bin/curl -o /dev/null --noproxy \* --data @/mnt/sysimage/root/install.post.log --silent 'http://192.168.122.1:8448/unattended/built?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  elif [ -x /usr/bin/wget ]; then
    /usr/bin/wget -q -O /dev/null --no-proxy --method POST --header 'Content-Type: text/plain' --body-file=/mnt/sysimage/root/install.post.log 'http://192.168.122.1:8448/unattended/built?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  else
    wget -q -O /dev/null --header 'Content-Type: text/plain' 'http://192.168.122.1:8448/unattended/built?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  fi
  else
  echo "calling home: build failed!"
    if [ -x /usr/bin/curl ]; then
    /usr/bin/curl -o /dev/null --noproxy \* --data @/mnt/sysimage/root/install.post.log --silent 'http://192.168.122.1:8448/unattended/failed?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  elif [ -x /usr/bin/wget ]; then
    /usr/bin/wget -q -O /dev/null --no-proxy --method POST --header 'Content-Type: text/plain' --body-file=/mnt/sysimage/root/install.post.log 'http://192.168.122.1:8448/unattended/failed?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  else
    wget -q -O /dev/null --header 'Content-Type: text/plain' 'http://192.168.122.1:8448/unattended/failed?token=71cc00bb-4073-4f27-bf9f-3114847e1f63'
  fi
  fi

sync
%end
```

It needs more love and testing, curl needs Content-Type to be set, WIP. Needs https://github.com/theforeman/smart-proxy/pull/621

**Comment by lzap** [comment]
Do not merge, but you get the idea. Feel free to review.

**Comment by lzap** [comment]
@iNecas @timogoebel how about this? It needs small (cosmetic) patch in core and smart proxy patch mentioned above tho: https://github.com/theforeman/foreman/pull/6209 and https://github.com/theforeman/smart-proxy/pull/621

This is how it looks like:

```
if test -f /tmp/foreman_built; then
  echo "calling home: build is done!"
  if [ -x /usr/bin/curl ]; then
    /usr/bin/curl -o /dev/null --noproxy \* -H 'Content-Type: text/plain' --data @/mnt/sysimage/root/install.post.log --silent 'http://192.168.122.1:8448/unattended/built?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  elif [ -x /usr/bin/wget ]; then
    /usr/bin/wget -q -O /dev/null --no-proxy --method POST --header 'Content-Type: text/plain' --body-file=/mnt/sysimage/root/install.post.log 'http://192.168.122.1:8448/unattended/built?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  else
    wget -q -O /dev/null --header 'Content-Type: text/plain' 'http://192.168.122.1:8448/unattended/built?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  fi
else
  echo "calling home: build failed!"
  if [ -x /usr/bin/curl ]; then
    /usr/bin/curl -o /dev/null --noproxy \* -H 'Content-Type: text/plain' --data @/mnt/sysimage/root/install.post.log --silent 'http://192.168.122.1:8448/unattended/failed?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  elif [ -x /usr/bin/wget ]; then
    /usr/bin/wget -q -O /dev/null --no-proxy --method POST --header 'Content-Type: text/plain' --body-file=/mnt/sysimage/root/install.post.log 'http://192.168.122.1:8448/unattended/failed?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  else
    wget -q -O /dev/null --header 'Content-Type: text/plain' 'http://192.168.122.1:8448/unattended/failed?token=3bf9d0c8-14b8-49c8-ba02-a538ba9eff43'
  fi
fi
```

**Comment by timogoebel** [comment]
@lzap: Looks good to me, it's a bit hard to read but I guess we can't avoid that.

**Comment by iNecas** [comment]
Looks good to me approach-wise (had some comment in related PR). Have not tested though yet.

**Comment by lzap** [comment]
So @timogoebel is busy, how about @iNecas testing this feature? It needs small (cosmetic) patch in core and smart proxy patch mentioned above tho: https://github.com/theforeman/foreman/pull/6209 and https://github.com/theforeman/smart-proxy/pull/621

**Comment by mmoll** [comment]
@lzap tests fail

**Comment by lzap** [comment]
Oh rebased.

**Comment by lzap** [comment]
Anybody? Hello!

**Comment by mmoll** [comment]
merged, díky @lzap!

**Comment by lzap** [comment]
Thanks hopefully this does not break much stuff, good timing after 1.21 branching.

**Comment by tbrisker** [comment]
Looks like core change never got in, now after syncing regenerating the snapshots failed. I'll merge core now to unblock branching, but please make sure if there are changes done across repos that they are merged in coordination.

**Comment by mmoll** [comment]
I overlooked that there was a second core patch (indent macro) needed, anyway this one should be good.

**Pull request merged**